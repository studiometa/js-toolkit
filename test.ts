/* eslint-disable max-classes-per-file, no-unused-expressions */
import { Base, withMountWhenInView, withType } from '@studiometa/js-toolkit';
import type { BaseTypeParameter } from '@studiometa/js-toolkit';

class Bar<T extends BaseTypeParameter = BaseTypeParameter> extends Base<
  T & {
    $options: { bar: true };
  }
> {
  bar() {
    return true;
  }
}

class BarBar extends Bar<{ $options: { barBar: true } }> {
  mounted() {
    this.$options.bar;
    this.$options.barBar;
  }
}

class Foo extends withMountWhenInView<
  typeof Bar,
  { $el: HTMLAnchorElement; $options: { foo: true } }
>(Bar) {
  mounted() {
    this.$el.href;
    this.__observer;
    this.$options.foo;
    this.$options.bar;
    this.$options.intersectionObserver;
    this.bar();
  }
}

class Baz extends withType<typeof Foo, { $options: { baz: true } }>(Foo) {
  mounted() {
    this.$el.href;
    this.bar();
    this.$options.foo;
    this.$options.bar;
    this.$options.intersectionObserver;
    this.$options.custom;
    this.$options.baz;
  }
}

export class Boz extends withType<typeof Baz, { $options: { boz: string } }>(Baz) {
  mounted() {
    this.$options.foo;
    this.$options.baz;
    this.$options.boz;
  }
}
