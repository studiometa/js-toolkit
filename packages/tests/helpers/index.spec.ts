import { test, expect } from 'bun:test';
import * as helpers from '../../js-toolkit/helpers/index.js';

test('helpers exports', () => {
  expect(Object.keys(helpers)).toMatchSnapshot();
});
