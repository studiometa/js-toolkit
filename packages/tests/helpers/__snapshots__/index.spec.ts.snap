// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers exports 1`] = `
[
  "createApp",
  "getClosestParent",
  "getDirectChildren",
  "getInstanceFromElement",
  "importOnInteraction",
  "importOnMediaQuery",
  "importWhenIdle",
  "importWhenPrefersMotion",
  "importWhenVisible",
  "isDirectChild",
]
`;

exports[`helpers exports 2`] = `
[
  "createApp",
  "getClosestParent",
  "getDirectChildren",
  "getInstanceFromElement",
  "importOnInteraction",
  "importOnMediaQuery",
  "importWhenIdle",
  "importWhenPrefersMotion",
  "importWhenVisible",
  "isDirectChild",
]
`;

exports[`helpers exports 3`] = `
[
  "createApp",
  "getClosestParent",
  "getDirectChildren",
  "getInstanceFromElement",
  "importOnInteraction",
  "importOnMediaQuery",
  "importWhenIdle",
  "importWhenPrefersMotion",
  "importWhenVisible",
  "isDirectChild",
]
`;

exports[`helpers exports 4`] = `
[
  "createApp",
  "getClosestParent",
  "getDirectChildren",
  "getInstanceFromElement",
  "importOnInteraction",
  "importOnMediaQuery",
  "importWhenIdle",
  "importWhenPrefersMotion",
  "importWhenVisible",
  "isDirectChild",
]
`;

exports[`helpers exports 5`] = `
[
  "createApp",
  "getClosestParent",
  "getDirectChildren",
  "getInstanceFromElement",
  "importOnInteraction",
  "importOnMediaQuery",
  "importWhenIdle",
  "importWhenPrefersMotion",
  "importWhenVisible",
  "isDirectChild",
]
`;

exports[`helpers exports 6`] = `
[
  "createApp",
  "getClosestParent",
  "getDirectChildren",
  "getInstanceFromElement",
  "importOnInteraction",
  "importOnMediaQuery",
  "importWhenIdle",
  "importWhenPrefersMotion",
  "importWhenVisible",
  "isDirectChild",
]
`;

exports[`helpers exports 7`] = `
[
  "createApp",
  "getClosestParent",
  "getDirectChildren",
  "getInstanceFromElement",
  "importOnInteraction",
  "importOnMediaQuery",
  "importWhenIdle",
  "importWhenPrefersMotion",
  "importWhenVisible",
  "isDirectChild",
]
`;

exports[`helpers exports 8`] = `
[
  "createApp",
  "getClosestParent",
  "getDirectChildren",
  "getInstanceFromElement",
  "importOnInteraction",
  "importOnMediaQuery",
  "importWhenIdle",
  "importWhenPrefersMotion",
  "importWhenVisible",
  "isDirectChild",
]
`;

exports[`helpers exports 9`] = `
[
  "createApp",
  "getClosestParent",
  "getDirectChildren",
  "getInstanceFromElement",
  "importOnInteraction",
  "importOnMediaQuery",
  "importWhenIdle",
  "importWhenPrefersMotion",
  "importWhenVisible",
  "isDirectChild",
]
`;
